vrs/utils/PixelFrameJpeg.cpp

```diff
@@ -40,35 +40,54 @@ bool PixelFrame::readJpegFrame(RecordReader* reader, const uint32_t sizeBytes) {
   return readJpegFrame(jpegBuf);
 }
 
-bool PixelFrame::readJpegFrame(const vector<uint8_t>& jpegBuf, bool decodePixels) {
-  // setup libjpeg
-  struct jpeg_decompress_struct cinfo;
-  struct jpeg_error_mgr jerr;
-  cinfo.err = jpeg_std_error(&jerr);
-  jpeg_create_decompress(&cinfo);
-  jpeg_mem_src(&cinfo, jpegBuf.data(), jpegBuf.size());
+static bool readJpegFrameHelper(PixelFrame& frame, struct jpeg_decompress_struct& cinfo, bool decodePixels) {
   jpeg_read_header(&cinfo, TRUE);
   jpeg_start_decompress(&cinfo);
   if (cinfo.num_components == 1) {
     cinfo.out_color_space = JCS_GRAYSCALE;
-    init(PixelFormat::GREY8, cinfo.image_width, cinfo.image_height);
+    frame.init(PixelFormat::GREY8, cinfo.image_width, cinfo.image_height);
   } else {
     cinfo.out_color_space = JCS_RGB;
-    init(PixelFormat::RGB8, cinfo.image_width, cinfo.image_height);
+    frame.init(PixelFormat::RGB8, cinfo.image_width, cinfo.image_height);
   }
   if (decodePixels) {
     // decompress row by row
-    uint8_t* rowPtr = frameBytes_.data();
+    uint8_t* rowPtr = frame.getBuffer().data();
     while (cinfo.output_scanline < cinfo.output_height) {
       jpeg_read_scanlines(&cinfo, &rowPtr, 1);
-      rowPtr += imageSpec_.getStride();
+      rowPtr += frame.getSpec().getStride();
     }
   }
   jpeg_finish_decompress(&cinfo);
   jpeg_destroy_decompress(&cinfo);
   return true;
 }
 
+bool PixelFrame::readJpegFrame(const vector<uint8_t>& jpegBuf, bool decodePixels) {
+  // setup libjpeg
+  struct jpeg_decompress_struct cinfo;
+  struct jpeg_error_mgr jerr;
+  cinfo.err = jpeg_std_error(&jerr);
+  jpeg_create_decompress(&cinfo);
+  jpeg_mem_src(&cinfo, jpegBuf.data(), jpegBuf.size());
+  return readJpegFrameHelper(*this, cinfo, decodePixels);
+}
+
+bool PixelFrame::readJpegFrameFromFile(const std::string& path, bool decodePixels) {
+  FILE* infile = fopen(path.c_str(), "rb");
+  if (infile == nullptr) {
+    return false;
+  }
+
+  // setup libjpeg
+  struct jpeg_decompress_struct cinfo;
+  struct jpeg_error_mgr jerr;
+  cinfo.err = jpeg_std_error(&jerr);
+  jpeg_create_decompress(&cinfo);
+  jpeg_stdio_src(&cinfo, infile);
+  return readJpegFrameHelper(*this, cinfo, decodePixels);
```

Comment:
I'm seeing an `fopen` but no `fclose`. Am I missing something?...

Reply:
good catch, will fix now

Reply:
@georges-berenger do you have any other feedback? otherwise, would you be able to merge the request? thanks so much for reviewing!